monitoring/daily_report.py

```
@@ -0,0 +1,652 @@
+        if not match:
+            stats['error'] = 'Invalid database URL'
+            return stats
+
+        user, password, host, port, database = match.groups()
+
+        # Query for stats from unified table (aggregate counts only, no PII)
+        queries = {
+            'total': "SELECT COUNT(*) FROM newsletter_unified",
+            'confirmed': "SELECT COUNT(*) FROM newsletter_unified WHERE status = 'confirmed'",
+            'pending': "SELECT COUNT(*) FROM newsletter_unified WHERE status = 'pending'",
+            'new_today': f"SELECT COUNT(*) FROM newsletter_unified WHERE created_at::date = '{date_str}'",
```

Ruff (0.15.5)

[error] Possible SQL injection vector through string-based query construction

(S608)